lib/core/HTTP.ts

```diff
@@ -121,4 +121,53 @@ export default class HTTP {
       return apiResponse.model<T>($clazz as new() => T)
     }
   }
+
+  public onRequest (fn: Function): void {
+    this.context.$axios.interceptors.request.use((config: AxiosRequestConfig) => {
+      return fn(config) || config
+    }, undefined)
+  }
+
+  public onResponse (fn: Function): void {
+    this.context.$axios.interceptors.response.use((response: AxiosResponse) => {
+      return fn(response) || response
+    }, undefined)
+  }
+
+  public onResponseError (fn: Function): void {
+    this.context.$axios.interceptors.response.use(undefined, (error: AxiosError) => {
+      return fn(error) || Promise.reject(error)
+    })
+  }
+
+  public onRequestError (fn: Function): void {
+    this.context.$axios.interceptors.request.use(undefined, (error: AxiosError) => {
+      return fn(error) || Promise.reject(error)
+    })
+  }
+
+  public onError (fn: Function): void {
+    this.onResponseError(fn)
+    this.onRequestError(fn)
+  }
+
+  public installDebugInterceptors (): void {
+    this.onRequestError((error: AxiosError) => {
+      console.info('Request error:', error)
+    })
+
+    this.onResponseError((error: AxiosError) => {
+      console.info('Response error:', error)
+    })
+
+    this.onResponse((response: AxiosResponse) => {
+      console.info(
```

Comment:
We should avoid using `console`, [Cosnola](https://github.com/nuxt-contrib/consola) would be a great replacement

Reply:
Right, I'll do that!